ci.md

```diff
@@ -0,0 +1,191 @@
+
+For libSBML, the YAML files are generally structured into:
+- define strategy matrix
+- checkout the repository
+- dependency installation and system setup
+- configure with `cmake`
+- build
+- test with `ctest`
+
+The nightly build additionally uploads some artefacts at the end.
+
+### Dependency installation
+Dependency installation varies across operating systems, and this is implemented via conditions on the strategy matrix entry, e.g.
+```yaml
+      - name: Do something Windows-specific
+        if: matrix.platform == 'windows-latest'
+        run: # some windows-specific commands
+```
+For convenience, we set up the `ninja` generator for all runs using the [`seanmiddleditch/gha-setup-ninja`](https://github.com/marketplace/actions/install-ninja-build-tool) action.
+
+On Unix-based systems, missing dependences are typically installed via standard package managers (`brew`, `apt-get`, `yum`). In `manylinux2010`, recent enough versions of `cmake` (via `pip`) and `swig` (compile from source) are installed in a non-standard way.
+
+Windows builds require a precompiled dependencies folder, which is downloaded from SourceForge and cached, as well as a "manual" swig set-up by the CI. Additionally, we use the [`ilammy/msvc-dev-cmd`](https://github.com/marketplace/actions/enable-developer-command-prompt) action to ensure `msbuild` remains accessible to `cmake` (it's not by default because we use the `ninja` generator).
+
+### CMake configuration
+All runs use the `ninja` generator to configure the `cmake` build. This has the advantage of automatically parallelising the build under the hood.
+
+We further use `ccache` to take advantage of several runs repeatedly calling the same compilation command.
+
+Windows builds are compiled using the MSVC static runtime (`-DWITH_STATIC_RUNTIME`).
+
+### Python
+
+Unix-based builds use the `PYTHON_USE_DYNAMIC_LOOKUP` option, which allows libSBML binaries to dynamically link to available Python libraries, instead of being tied to the Python libraries of the OS the binaries were compiled on.
+
+### R package and bindings
+Unix-based systems additionally build a binary R package in two steps: first the compiliation is configured to skip building the R binaries and create a source package instead (`-DWITH_CREATE_R_SOURCE=ON -DWITH_SKIP_R_BINARY=ON`). The binary R package is the created from the source package using the `R CMD INSTALL`.See the documentation for [creating R source packages and binary packages](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Building-binary-packages).
+
+# Using the CI system to make a libSBML release
```

Comment:
Not sure we need this section here (or at all)? It might still be helpful as a basis to describe how the release system has changed in the future (e.g. in a publication...)